tools/apicoverage/apicoverage.go

```diff
@@ -187,57 +186,65 @@ func getRelevantLogs(fields []string) *string {
-	tp := []testgrid.TestProperty{testgrid.TestProperty{Name: apiCoverage, Value: percentCovered}}
-	tc = append(tc, testgrid.TestCase{Name: tcName, Properties: testgrid.TestProperties{Property: tp}, Fail: false})
+	testCase := junit.TestCase{Name: tcName}
+	testCase.AddProperty(apiCoverage, fmt.Sprintf("%f", percentCovered))
+	suite.AddTestCase(&testCase)
+
+	fmt.Println("add test case")
 
 	for key, value := range covered {
-		tp := []testgrid.TestProperty{testgrid.TestProperty{Name: apiCoverage, Value: float32(value)}}
-		tc = append(tc, testgrid.TestCase{Name: tcName + "/" + key, Properties: testgrid.TestProperties{Property: tp}, Fail: false})
+		coveredTestCase := junit.TestCase{Name: tcName + "/" + key}
+		coveredTestCase.AddProperty(apiCoverage, fmt.Sprintf("%d", value))
+		suite.AddTestCase(&coveredTestCase)
 	}
 
 	for key, value := range notCovered {
-		tp := []testgrid.TestProperty{testgrid.TestProperty{Name: apiCoverage, Value: float32(value)}}
-		tc = append(tc, testgrid.TestCase{Name: tcName + "/" + key, Properties: testgrid.TestProperties{Property: tp}, Fail: true})
+		notCoveredTestCase := junit.TestCase{Name: tcName + "/" + key}
+		notCoveredTestCase.AddProperty(apiCoverage, fmt.Sprintf("%d", value))
+		notCoveredTestCase.SetFailure("failure")
+		suite.AddTestCase(&notCoveredTestCase)
 	}
-	return tc
 }
 
 func createTestgridXML(coverage *OverallAPICoverage, artifactsDir string) {
-	tc := createCases(overallRoute, coverage.RouteAPICovered, coverage.RouteAPINotCovered)
-	tc = append(tc, createCases(overallConfig, coverage.ConfigurationAPICovered, coverage.ConfigurationAPINotCovered)...)
-	tc = append(tc, createCases(overallService, coverage.ServiceAPICovered, coverage.ServiceAPINotCovered)...)
-	ts := testgrid.TestSuite{TestCases: tc}
+	suite := junit.NewSuite("")
+	addTestCases(overallRoute, coverage.RouteAPICovered, coverage.RouteAPINotCovered, suite)
+	addTestCases(overallConfig, coverage.ConfigurationAPICovered, coverage.ConfigurationAPINotCovered, suite)
+	addTestCases(overallService, coverage.ServiceAPICovered, coverage.ServiceAPINotCovered, suite)
 
-	if err := testgrid.CreateXMLOutput(ts, artifactsDir, "apicoverage"); err != nil {
+	suites := junit.NewSuites()
+	suites.AddSuite(suite)
+
+	if err := testgrid.CreateXMLOutput(suites, "apicoverage"); err != nil {
 		log.Fatalf("Cannot create the xml output file: %v", err)
 	}
 }
 
 func main() {
-
 	artifactsDir := flag.String("artifacts-dir", "./artifacts", "Directory to store the generated XML file")
 	serviceAccount := flag.String("service-account", os.Getenv("GOOGLE_APPLICATION_CREDENTIALS"), "JSON key file for service account to use")
 	flag.Parse()
 
-	// Read the latest-build.txt file to get the latest build number
-	ctx := context.Background()
-	num, err := gcs.GetLatestBuildNumber(ctx, logDir, *serviceAccount)
+	// Explicit authenticate with gcs Client
+	prow.Initialize(*serviceAccount)
+
+	jobToMonitor := prow.NewJob(targetJob)
+	jobToMonitor.Type = prow.PostsubmitJob
```

Comment:
do we need to set this?

Reply:
prow figures out gcs job path based on job type, setting this so that we don't need to set gcs job path later.

Reply:
So, that has an intended side-effect. Maybe we should add the type as an arg to newjob() to be more explicit that this is needed.

Reply:
I thought about that too, there is another element in Job struct that can be used to set gcs job path instead of computing gcs job path from Job Type, so in this case Type becomes optional. 
Sorry for the confusion, I have added comment around `Job` struct clarifying this.